### ISSUE TITLE
Problems with typed array

### ISSUE THREAD

**Issue by thigg** [exaV/screeps-kotlin-types#42]
To be able to supply a list of RoomObjects to findClosestByPath I had to manually map it to the type.

Either I didn't understand very well, how kotlin does typing here, kkotlin messes up the type inference, or this is an issue with the definition.

```
val targets: Array<RoomObject> = (room.find(FIND_SOURCES) +
            room.find(FIND_DROPPED_ENERGY) +
            room.find(FIND_STRUCTURES).filter { s ->
                when (s.structureType) {
                    STRUCTURE_CONTAINER, STRUCTURE_STORAGE -> true
                    else -> false
                }
            }).map { a -> a as RoomObject }.toTypedArray()
    val target :RoomObject? = pos.findClosestByPath<RoomObject>(targets)
```

**Comment by Jomik** [comment]
That's probably because the filter is using Structure and not RoomObject. You can avoid the map, by casting

**Comment by AWRyder** [comment]
I believe the issue is, as Jomik said, not the cast to typed array but the cast to a typed array using a specific generic. Try adding a generic on find so that you get a collection of the type you want.
find<RoomObject>(...) for example.

**Comment by thigg** [comment]
No, I tried `pos.findClosestByPath<RoomObject>((targets + creepTargets).map { a -> a as RoomObject }.toTypedArray())` and it does not work without the map, at least intellij doesnt like it then.

**Comment by exaV** [comment]
You should be able to write the following:

```
 val targets :  Array<RoomObject> =
                room.find(FIND_SOURCES) +
                room.find(FIND_DROPPED_ENERGY) +
                room.find(FIND_STRUCTURES).filter { s ->
                    when (s.structureType) {
                        STRUCTURE_CONTAINER, STRUCTURE_STORAGE -> true
                        else -> false
                    }
                }
val target :RoomObject? = pos.findClosestByPath(targets)
```
It is a bit annoying to write the type of targets down. This is an error on our part because `pos.findClosestByPath()` does not accept a subtype of RoomObject as an argument, even though it should! We'll fix it soon.

**Comment by exaV** [comment]
this should be fixed with the upgrade to Kotlin 2.0
